packages/esm-covid-app/src/home.component.tsx

```diff
@@ -1,16 +1,158 @@
+
+const HomeComponent = () => {
+  const [headers, setHeaders] = useState([]);
+  const [rows, setRows] = useState([]);
+  const [reportId, setReportId] = useState('mother_hiv_status');
+  const [ptrackerId, setPtrackerId] = useState('12345A232567');
+  const [personUuid, setPersonUuid] = useState('bd49d697-b1de-49b9-95c2-6031fb1375fd');
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
+
+  const fetchData = async () => {
+    if (!startDate || !endDate) {
+      console.error('Start date and end date must be provided.');
+      return;
+    }
+
+    try {
+      const url = `${BASE_WS_API_URL}?report_id=${reportId}&ptracker_id=${ptrackerId}&person_uuid=${personUuid}&start_date=${startDate}&end_date=${endDate}`;
+      const response = await openmrsFetch(url, {
+        method: 'GET',
+        headers: {
+          Authorization: 'Basic ' + btoa('root:12345678'),
+        },
+        credentials: 'include',
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const data = await response.json();
+
+      if (data.results.length === 0) {
+        setHeaders([]);
+        setRows([]);
+        return;
+      }
+
+      // Extract headers from the first record
+      const formattedHeaders = data.results[0].record.map((column) => ({
+        key: column.column,
+        header: column.column,
+      }));
+
+      const formattedRows = data.results.map((result) => {
+        const row = { id: result.serialId.toString() };
+        result.record.forEach((column) => {
+          row[column.column] = column.value === null ? '-' : column.value;
+        });
+        return row;
+      });
+
+      setHeaders(formattedHeaders);
+      setRows(formattedRows);
+    } catch (error) {
+      console.error('Error fetching data:', error);
+    }
+  };
+
+  const handleStartDateChange = (event) => {
+    const date = event[0] ? event[0].toISOString().split('T')[0] : '';
+    setStartDate(date);
+  };
+
+  const handleEndDateChange = (event) => {
+    const date = event[0] ? event[0].toISOString().split('T')[0] : '';
+    setEndDate(date);
+  };
 
-const Homecomponent = () => {
   return (
-    <div>
-      <OHRIWelcomeSection title="COVID-19 Cases" />
-      <CovidSummaryTiles />
-      <CovidHomePatientTabs />
+    <div className={styles.homeContainer}>
+      <OHRIWelcomeSection title="Reporting Demo" />
+      <form
+        className={styles.form}
+        onSubmit={(e) => {
+          e.preventDefault();
+          fetchData();
+        }}
+      >
+        <div className={styles.datePickerContainer}>
+          <Dropdown
+            id="report-dropdown"
+            titleText="Select Report"
+            label="Select a report to display"
+            items={[
+              { id: 'ecabd559-14f6-4c65-87af-1254dfdf1304', text: 'Covid-19 Report' },
+              { id: '3ffa5a53-fc65-4a1e-a434-46dbcf1c2de2', text: 'HTS Report' },
```

Comment:
Lets add these uuids in the config

Reply:
made the requested changes. kindly check new commit